src/Features/VisualBasic/Portable/CodeLens/VisualBasicDisplayInfoService.vb

```diff
@@ -144,5 +144,54 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeLens
 
             Return symbolName
         End Function
+
+        Private Shared Function GetRootNamespace(symbol As ISymbol) As String
+            Dim containingNamespace = symbol.ContainingNamespace
+
+            Dim rootName = String.Empty
+            If containingNamespace IsNot Nothing Then
+                Dim name = containingNamespace.ToDisplayString()
+                If String.Compare(name, "Global", StringComparison.OrdinalIgnoreCase) <> 0 Then
+                    rootName = name
+                End If
+            End If
+
+            Return rootName
+        End Function
+
+        Public Function ConstructFullName(semanticModel As SemanticModel, methodNode As SyntaxNode) As String Implements ICodeLensDisplayInfoService.ConstructFullName
+            Dim isTopLevelNodeAClass = True
+            Dim parent = methodNode.Parent
+            Dim name = semanticModel.GetDeclaredSymbol(methodNode).Name
+
+            While parent IsNot Nothing
```

Comment:
any time we're manually constructing a name, that is a bad smell.  We created SymbolDisplay precisely for the purpose of having a single unified location for building names.